### DISCUSSION TITLE
Coupled DeepONet PDE with Hard Boundary Conditions

### DISCUSSION THREAD

**Discussion by schrodingersket** [lululxvi/deepxde#1411] [Q&A]
Hi there,

~~[This paper](https://arxiv.org/pdf/2111.05512.pdf) discusses an implementation of DeepONets which supports multiple function outputs and enforcing hard Dirichlet boundary conditions, but I couldn't find an example of that implementation in https://github.com/lu-group/deeponet-fno. Is there support for this in DeepXDE?~~ Looks like there [pull request](https://github.com/lululxvi/deepxde/pull/1410) open for this functionality. 

One of my DeepONet boundary conditions involves the function which sampled from the input space. Is there a way to expose this function to `apply_output_transform` in a manner similar to the way that `DirichletBC` and `IC` do in order to enforce that as a hard constraint? The constraint is essentially just `u(L) + v(L) = 1.0`, where `v` is the function from the input space. Also, is it possible to differentiate these functions within a `PDE` function? I tried using `dde.grad.jacobian` to differentiate with respect to the inputs, but that didn't quite seem to work (and returned a `None` value).

Thanks in advance! I'm happy to share details of the particular problem I'm working on if that's helpful, but I didn't want to distract from the question right off the bat.

**Comment by lululxvi** [comment]
You can see examples here https://deepxde.readthedocs.io/en/latest/demos/operator.html#pi-deeponet

**Reply by schrodingersket** [reply]
I did look through those examples; however, they do not seem to indicate whether `apply_output_transform` provides auxiliary functions from the branch network function space as an argument. I dug into the source code a bit and it seems that is not provided as a function argument, but I wanted to verify that to be the case.

Thanks again!
